### PR TITLE
[SPARK-19852][PYSPARK][ML] Update Python API setHandleInvalid for StringIndexer

### PR DESCRIPTION
**Pull request by VinceShieh**
## What changes were proposed in this pull request?

This PR is to maintain API parity with changes made in SPARK-17498 to support a new option
'keep' in StringIndexer to handle unseen labels with pyspark

## How was this patch tested?
existing tests
testing is done with new doctests

Signed-off-by: VinceShieh <vincent.xie@intel.com>

### CODE REVIEW

**Comment by jkbradley** [comment]
Thanks for the PR!  I just merged the fix for https://issues.apache.org/jira/browse/SPARK-11569 which will affect this PR.  Would you mind updating this PR to include SPARK-11569's handling of null values?

**Comment by VinceShieh** [comment]
Sure. No problem!

File: python/pyspark/ml/feature.py [hunk 1]
```diff
@@ -1936,6 +1935,14 @@ class StringIndexer(JavaEstimator, HasInputCol, HasOutputCol, HasHandleInvalid,
...
     ...     key=lambda x: x[0])
     [(0, 'a'), (1, 'b'), (2, 'c'), (3, 'a'), (4, 'a'), (5, 'c')]
+    >>> testData2 = sc.parallelize([Row(id=0, label="a"), Row(id=1, label="d"),
+    ...     Row(id=2, label=None)], 2)
+    >>> dfKeep= spark.createDataFrame(testData2)
+    >>> modelKeep = stringIndexer.setHandleInvalid("keep").fit(stringIndDf)
+    >>> tdK = modelKeep.transform(dfKeep)
+    >>> sorted(set([(i[0], i[1]) for i in tdK.select(tdK.id, tdK.indexed).collect()]),
+    ...     key=lambda x: x[0])
+    [(0, 0.0), (1, 3.0), (2, 3.0)]
```
Comment (yanboliang) [inline python/pyspark/ml/feature.py]: Could you move the newly added test to ```tests.py```? We keep the basic doc tests here both for test and example, other tests should be placed at ```tests.py```. Thanks.

**Review by yanboliang — COMMENTED** [review]
LGTM except for one minor comment. Thanks.

File: python/pyspark/ml/feature.py (hunk 1 above)
Comment (holdenk) [inline python/pyspark/ml/feature.py]: +1

Comment (VinceShieh) [inline python/pyspark/ml/feature.py]: Okay. Sure @yanboliang @holdenk

Comment (yanboliang) [inline python/pyspark/ml/feature.py]: @VinceShieh Do you have time to update this PR? We would like to get this in 2.2. Thanks.

**Comment by holdenk** [comment]
Gentle ping to @VinceShieh to see if you have a chance to update this with the test movement & merge in the latest from master branch?

**Comment by holdenk** [comment]
gentle re-ping

**Comment by gatorsmile** [comment]
We are closing it due to inactivity. please do reopen if you want to push it forward. Thanks!

**Pull request closed without merging**